plugin/openstudio/lib/dialogs/SurfaceMatchingDialog.rb

```diff
@@ -136,13 +136,32 @@ def intersect(selection)
 
       # iterate through spaces to create intersecting geometry
       spaces.each do |space|
-        entity = space.entity
-        entity.entities.intersect_with(true, entity.transformation, entity.entities.parent, entity.transformation, false, selection.to_a)
+		entity = space.entity
+		before_faces = entity.entities.grep(Sketchup::Face) #create array of faces before the intersect
+		entity.entities.intersect_with(true, entity.transformation, entity.entities.parent, entity.transformation, false, selection.to_a)
+        edges = entity.entities.grep(Sketchup::Edge)
+		edges.each {|edge| edge.find_faces} #Heal empty loops (no face) after intersection, this may not be necessary?
+
+		#Ski90Moo:This will find faces with AttributeDictionaries copied from the base face (this is causing the faces to be added to the drawing_interface but a new model surface object is not created because all the copies point to an existing surface object.) Rough work around solution is to delete the dictionary, delete the sketchup entity, and redraw it.  This prompts the downstream observer to create a new model surface object and add the sketchup entities to the drawing_interface.  There is probably a more direct solution without having to delete and redraw, but I do not fully understand how the observer works.
```

Comment:
So the idea is basically to trick SketchUp into thinking the surfaces after intersection are new?  I think you identified the root cause of the issue but I do wonder if there is a better solution.  Do you have a test model you could upload that demonstrates the bad behavior?  You can upload it to a comment here if you rename to have a *.txt extension

Reply:
Yes, I agree.  Utlimately, we need to get the observer to recognize the new surfaces as actually new, unique entities such that is adds them to the drawing_interface register with unique drawing_interface ids and also creates the corresponding OpenStudio surface object.  Here is the test osm file: 
[test.txt](https://github.com/openstudiocoalition/openstudio-sketchup-plugin/files/10366711/test.txt)

Reply:
Thanks @Ski90Moo, I can reproduce the issue with this test model

Reply:
Thanks for taking a look at this @macumber.

Reply:
Does this issue happen every time for you on this model @Ski90Moo?  I am having trouble reproducing it tonight for some reason.

Reply:
Yes, it happens every time, however, it is very random.  Sometimes the intersect will successfully create several OS model object surfaces, other times it will only be successful with a single surface.  The best way to see the problem (missing surfaces) is to do the intersect, click save, and then reopen the model in sketchup.  All of the face entities that didn’t get added to the model as surface objects will be gone.

Reply:
Do you need more info on this @macumber?

Reply:
I got a little stuck on this.  I originally had seen the missing surfaces but now am not able to reproduce (using SketchUp Pro 2022 and OS App 1.4.0 or 1.5.0).  I am using this one liner to calculate the area of surfaces before and after the intersection: `OpenStudio::Plugin.model_manager.model_interface.openstudio_model.getSurfaces.inject(0) {|sum, s| sum += s.grossArea }`

I do see some missed matching data but the intersections are all looking clean.

Reply:
Ok, I tried with SketchUp Pro 2021, OS App 1.5.0, and OS Skp Plugin 1.4.0 and it is reproducible again!

Reply:
Running in SU 2022 produces 34 "no swap" messages, SU 2021 produces 9.

Reply:
Just a note but the "Intersect Space Geometry" user script also results in missing surfaces.

Reply:
I'm only seeing the missing faces in SU 2021, which version are you using @Ski90Moo?  If it is just a bug in SU 2021, I think I'd rather just ignore this part and focus on the better matching which affects all versions.  What do you think?

Reply:
Ok, that makes sense.  I am using 2021 (mainly because of the renaming spaces in inspector issue).  I'll do some testing with 2022 and see if it is still an issue.  Otherwise, I agree, the better matching should be the focus for now.

Reply:
I addition to checking the area before and after

`OpenStudio::Plugin.model_manager.model_interface.openstudio_model.getSurfaces.inject(0) {|sum, s| sum += 1 }` returns 96 after intersection (in case of self intersecting polygon).  It returns 97 for correct intersection.  There are 62 surfaces to begin with.

Reply:
I found another surface that can fail to get broken into multiple pieces, unfortunately this occurs in SketchUp 2023 as well. 

![image](https://user-images.githubusercontent.com/1327850/226200169-3f885e4b-d8f5-4e39-9e43-fb23f0b521bb.png)

Reply:
And another failed intersection

![image](https://user-images.githubusercontent.com/1327850/226200838-d611250c-8249-416c-b376-7ae40a12a25c.png)

Reply:
Can you send me the test model?